src/main/java/cubicchunks/server/ServerCubeCache.java

```diff
@@ -405,6 +419,11 @@ public void unloadCube(int cubeX, int cubeY, int cubeZ) {
 		if (cubeIsNearSpawn(cubeX, cubeY, cubeZ)) {
 			return;
 		}
+		
+		// Do not unload cubes which are required for generating currently loaded cubes.
+		if (dependencyManager.isRequired(cubeX, cubeY, cubeZ)) {
+			return;
```

Comment:
I did some semi-formal data-flow analysis and I have not found any leaks. Running around a map for ~30min did not cause any increase in RAM usuage either.